Source/NSubstitute/Core/CallResults.cs

```diff
@@ -6,12 +6,18 @@ namespace NSubstitute.Core
     public class CallResults : ICallResults
     {
         readonly ICallInfoFactory _callInfoFactory;
+        private readonly bool _skipVoidMethods;
         ConcurrentQueue<ResultForCallSpec> _results;
 
-        public CallResults(ICallInfoFactory callInfoFactory)
+        public CallResults(ICallInfoFactory callInfoFactory, bool skipVoidMethodsForPerformance)
```

Comment:
I'm sorry for being intrusive here. I cannot get the purpose of the `SkipVoidMethod` method. Could you point me to the AF code. As I imagined AF will have a custom handler to set out/ref parameters. In case of void method `CallResults` handler will be skiped. I pushed a PR with the change  https://github.com/Zvirja/NSubstitute/pull/2/files